### PR TITLE
Bug 2045927: Add proxy for ironic-agent.service

### PR DESCRIPTION
**Pull request by zhouhao3**
Avoid the issue that ironic agent image cannot be downloaded due to network proxy.

See https://github.com/openshift/installer/issues/5552 for details.

Signed-off-by: Zhou Hao <zhouhao@fujitsu.com>

### CODE REVIEW

File: pkg/ignition/service_config.go [hunk 1]
```diff
@@ -35,13 +35,17 @@ Description=Ironic Agent
 After=network-online.target
 Wants=network-online.target
 [Service]
-TimeoutStartSec=0
+Environment="HTTP_PROXY=%s"
+Environment="HTTPS_PROXY=%s"
+Environment="NO_PROXY=%s"
+TimeoutStartSec=10
```
Comment (dtantsur) [inline pkg/ignition/service_config.go:41]: Why was this value changed? If needed for other reasons, I'd prefer a separate PR for this.

**Comment by dtantsur** [comment]
/ok-to-test

You'll need to attach a valid bug to merge this change at this point.

File: pkg/ignition/service_config.go (hunk 1 above)
Comment (zhouhao3) [reply pkg/ignition/service_config.go]: ok , I will open a separate PR for this.

**Comment by zhouhao3** [comment]
>You'll need to attach a valid bug to merge this change at this point.

We are preparing BZ, I will update later.

**Comment by zhouhao3** [comment]
@hardys @andfasano PTAL

**Pull request merged**